src/include/ipc/topology.h

```diff
@@ -156,6 +166,7 @@ enum sof_volume_ramp {
 	SOF_VOLUME_WINDOWS_FADE,
 	SOF_VOLUME_WINDOWS_NO_FADE,
 };
+#endif
```

Comment:
Ouch, this is a problem. We have these definitions also in Linux kernel and we can't have #ifdef for IPC version. So if we need separate definitions, we need to use a different name.

Reply:
@singalsu , @kv2019i  maybe no need to add #if, simply use new enum replace old ones, the reason is: sof use these as it is defined, I mean the sequence does not have impact on volume for both ipc3 and ipc4.

Reply:
@kv2019i The kernel treats this as generic u32

https://github.com/thesofproject/linux/blob/cffcac365eac0d60dc38959cadaa5078bc924b8e/sound/soc/sof/ipc3-topology.c#L119

https://github.com/thesofproject/linux/blob/cffcac365eac0d60dc38959cadaa5078bc924b8e/sound/soc/sof/ipc4-topology.c#L130

There is this header but it's not used. Could we remove it instead? 

https://github.com/thesofproject/linux/blob/cffcac365eac0d60dc38959cadaa5078bc924b8e/include/sound/sof/topology.h#L138

Reply:
@kv2019i As other fix approach could duplicate all ramp types to https://github.com/thesofproject/sof/blob/b608ec840c80b3e27bc8cc4d4d3a401f3929b396/src/audio/volume/peak_volume.h#L44 and add a conversion function from `enum ipc4_peak_volume_param` to `enum sof_volume_ramp`. What do you prefer?

Reply:
@singalsu  Can we have separate enums for IPC3 and IPC4 and also remove the unused enum in kernel? I think that would be the cleanest (no ifdefs in headers, no mismatch between definitions between kernel and FW).